integration-test/src/main/java/com/sequenceiq/it/cloudbreak/util/wait/service/database/DatabaseOperationChecker.java

```diff
@@ -0,0 +1,135 @@
+        try {
+            DatabaseServerV4Response database = waitObject.getEndpoint().getByCrn(crn);
+            if (database == null) {
+                throw new TestFailException(String.format("'%s' database was not found.", crn));
+            }
+            String name = database.getName();
+            Status status = database.getStatus();
+            LOGGER.info("Waiting for the '{}' state of '{}' '{}' database. Actual state is: '{}'", desiredStatus, name, crn, status);
+            if (isDeletionInProgress(status) || status == DELETE_COMPLETED) {
+                LOGGER.error("Database '{}' '{}' has been getting terminated (status:'{}'), waiting is cancelled.", name, crn, status);
+                throw new TestFailException(String.format("Database '%s' '%s' has been getting terminated (status:'%s'), waiting is cancelled.", name, crn,
+                        status));
+            }
+            if (waitObject.isFailed(status)) {
+                LOGGER.error("Database '{}' '{}' is in failed state (status:'{}'), waiting is cancelled.", name, crn, status);
+                throw new TestFailException(String.format("Database '%s' '%s' is in failed state. Status: '%s' statusReason: '%s'",
+                        name, crn, status, database.getStatusReason()));
+            }
+            if (desiredStatus.equals(status)) {
+                return true;
+            }
+        } catch (Exception e) {
+            StringBuilder builder = new StringBuilder("Database has been failed. Also failed to get database status: ")
+                    .append(System.lineSeparator())
+                    .append(e.getMessage())
+                    .append(System.lineSeparator())
+                    .append(e);
+            LOGGER.error(builder.toString());
+            throw new TestFailException(builder.toString());
+        }
+        return false;
+    }
+
+    @Override
+    public void handleTimeout(T waitObject) {
+        String crn = waitObject.getCrn();
+        try {
+            DatabaseServerV4Response database = waitObject.getEndpoint().getByCrn(crn);
+            if (database == null) {
+                throw new TestFailException(String.format("'%s' database was not found.", crn));
+            }
+            String name = database.getName();
+            Status status = database.getStatus();
+            throw new TestFailException(String.format("Wait operation timed out, database '%s' '%s' has been failed. Database status: '%s' "
+                    + "statusReason: '%s'", name, crn, status, database.getStatusReason()));
+        } catch (Exception e) {
+            StringBuilder builder = new StringBuilder("Wait operation timed out, database has been failed. Also failed to get database status: ")
+                    .append(System.lineSeparator())
+                    .append(e.getMessage())
+                    .append(System.lineSeparator())
+                    .append(e);
+            LOGGER.error(builder.toString());
```

Comment:
For pure logging please use the `LOGGER.error("message {} {}", arg1,, arg2)` format. I saw it other places as well. Please avoid manual string concatenation when possible.